src/scheduler/pbs_sched.cpp

```diff
@@ -540,26 +540,54 @@ static void
 connect_svrpool()
 {
 	int i;
-	svr_conn_t *svr_conns;
+	svr_conn_t *svr_conns_primary = NULL;
+	svr_conn_t *svr_conns_secondary = NULL;
 	int registered = 0;
+	int num_conf_svrs = get_num_servers();
 
 
 	while (clust_primary_sock < 0 || clust_secondary_sock < 0 || registered == 0) {
 		if (clust_primary_sock < 0) {
 			clust_primary_sock = pbs_connect(NULL);
+			/* clust_primary_sock later is used to fetch primary connections for all the servers */
 			if (clust_primary_sock < 0) {
 				/* wait for 2s for not to burn too much CPU, and then retry connection */
 				sleep(2);
 				continue;
 			}
 		}
 		clust_secondary_sock = pbs_connect(NULL);
+		/* clust_secondary_sock later is used to fetch secondary connections for all the servers */
 		if (clust_secondary_sock < 0) {
 			/* wait for 2s for not to burn too much CPU, and then retry connection */
 			sleep(2);
 			continue;
 		}
 
+		svr_conns_primary =  static_cast<svr_conn_t *>(get_conn_servers(clust_primary_sock));
+		svr_conns_secondary =  static_cast<svr_conn_t *>(get_conn_servers(clust_secondary_sock));
+
+		for (i = 0; i < num_conf_svrs; i++) {
+			if (svr_conns_primary[i].state == SVR_CONN_STATE_DOWN) {
+				clust_primary_sock = -1;
+				break;
+			}
+			if (svr_conns_primary[i].state == SVR_CONN_STATE_DOWN) {
+				clust_secondary_sock = -1;
+				break;
+			}
```

Comment:
The problem with putting the break where you have it is that you won't catch the case where both are down.  This is likely if a server crashed.  I'd change this to check if (prim == DOWN || sec == DOWN) then check individually for prim and sec setting the sockets to -1.  At the end of of the outer if statement, break.  Although in the situation where we do have one server which crashed, the other servers pick up for it.  Won't this cause us to wait for it to come back up before we continue?

Reply:
Thanks @bhroam. I have implemented as you suggested. Our goal is to wait till all servers come up.